code/modules/cm_marines/smartgun_mount.dm

```diff
@@ -370,6 +370,11 @@
 		if(fail)
 			to_chat(user, SPAN_WARNING("You can't install \the [src] here, something is in the way."))
 			return
+		if(istype(T, /turf/open))
+			var/turf/open/floor = T
+			if(!floor.allow_construction)
```

Comment:
More checks are required in this file as people are able to set it up this just stops them from screwing it into place and confusing people.